### DISCUSSION TITLE
Cannot use Prisma on WSL ( libssl is not found )

### DISCUSSION THREAD

**Discussion by kareemmahlees** [prisma/prisma#25232] [Q&A] [answered]
### Question

 I am trying to switch my workflow from Windows to WSL, but I am having some difficulties setting up Prisma correctly.
When I run `npx prisma generate --sql` the client is generated successfully, but after I start the application it immediately throws on the first query:
```
Error: Unable to require(`STRIPPED/node_modules/@prisma/engines/libquery_engine-debian-openssl-1.1.x.so.node`).
Prisma cannot find the required `libssl` system library in your system. Please install openssl-1.1.x and try again.

Details: libssl.so.1.1: cannot open shared object file: No such file or directory
    at one (STRIPPED/node_modules/prisma/build/index.js:284:116)
    at lWe (STRIPPED/node_modules/prisma/build/index.js:284:269)
```
Even executing the simplest `npx prisma version -v` throws the same error.
I have installed the following:
```
➜  apt list openssl -a
Listing... Done
openssl/focal-updates,focal-security,now 1.1.1f-1ubuntu2.23 amd64 [installed]
openssl/focal 1.1.1f-1ubuntu2 amd64
```
And here is the output of `openssl version -a`:
```
OpenSSL 3.3.1 4 Jun 2024 (Library: OpenSSL 3.3.1 4 Jun 2024)
built on: Tue Jun  4 12:53:04 2024 UTC
platform: linux-x86_64
options:  bn(64,64)
compiler: gcc-11 -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_BUILDING_OPENSSL -DNDEBUG
OPENSSLDIR: "/home/linuxbrew/.linuxbrew/etc/openssl@3"
ENGINESDIR: "/home/linuxbrew/.linuxbrew/Cellar/openssl@3/3.3.1/lib/engines-3"
MODULESDIR: "/home/linuxbrew/.linuxbrew/Cellar/openssl@3/3.3.1/lib/ossl-modules"
Seeding source: os-specific
CPUINFO: OPENSSL_ia32cap=0xfed83203078bffff:0x400684219c07a9
```

**Note**: one thing that I noticed is that `$LD_LIBRARY_PATH` is empty.

### How to reproduce (optional)

1. I just cloned a simple repo ( unfortunately it's private so I can't share, but I think this is not something specific to the repo ).
2. Ran `npm install`
3. And then simply started the app or ran `npx prisma version -a`, both yielded the same result.

### Expected behavior (optional)

Prisma should be able to find libssl, I know there is something that I am missing, so if you could correct me that would be awesome.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["typedSql"]
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}
```

```ts
// Add any relevant Prisma Client queries here
just simply running `npx prisma version -v`, or **Any** other query.
```

- OS: Windows WSL
- Database: PostgreSQL
- Node.js version: v22.7.0

- Run `prisma -v` to see your Prisma version and paste it:
```
prisma                  : 5.19.1
@prisma/client          : 5.19.1
Computed binaryTarget   : debian-openssl-1.1.x
Operating System        : linux
Architecture            : x64
Node.js                 : v22.7.0
Query Engine (Node-API) : E_CANNOT_RESOLVE_VERSION (at node_modules/@prisma/engines/libquery_engine-debian-openssl-1.1.x.so.node)
Schema Engine           : schema-engine-cli 69d742ee20b815d88e17e54db4a2a7a3b30324e3 (at node_modules/@prisma/engines/schema-engine-debian-openssl-1.1.x)
Schema Wasm             : @prisma/prisma-schema-wasm 5.19.1-2.69d742ee20b815d88e17e54db4a2a7a3b30324e3
Default Engines Hash    : 69d742ee20b815d88e17e54db4a2a7a3b30324e3
Studio                  : 0.502.0
Preview Features        : typedSql
```

**Comment by ludralph** [comment, marked as answer]
Hi @kareemmahlees 👋 

You can follow the suggestion in this previous [issue](https://github.com/prisma/prisma/issues/15505#issuecomment-1295905221) as the error message is similar.

1. Install OpenSSL 1.1.x: You need to have OpenSSL 1.1.x installed on your system. You can follow the steps below to manually install OpenSSL 1.1.x:
```
$ mkdir $HOME/opt && cd $HOME/opt
$ wget https://www.openssl.org/source/openssl-1.1.1o.tar.gz
$ tar -zxvf openssl-1.1.1o.tar.gz
$ cd openssl-1.1.1o
$ ./config && make && make test # some tests will fail, it's not an issue
$ mkdir $HOME/opt/lib
$ mv $HOME/opt/openssl-1.1.1o/libcrypto.so.1.1 $HOME/opt/lib/
$ mv $HOME/opt/openssl-1.1.1o/libssl.so.1.1 $HOME/opt/lib/
```
2. Set the `LD_LIBRARY_PATH`: After installing OpenSSL 1.1.x, you need to set the `LD_LIBRARY_PATH` environment variable to include the directory where the OpenSSL 1.1.x libraries are located:

export LD_LIBRARY_PATH=$HOME/opt/lib:$LD_LIBRARY_PATH

3. Verify the Installation: Ensure that the OpenSSL 1.1.x libraries are correctly installed and accessible by running:
```
ls $HOME/opt/lib
```
You should see libcrypto.so.1.1 and libssl.so.1.1 listed.

You may also find this [StackOverflow](https://stackoverflow.com/questions/72133316/libssl-so-1-1-cannot-open-shared-object-file-no-such-file-or-directory) post useful.

**Reply by kareemmahlees** [reply]
Thank you, this worked.
Although I was hoping it could find the openssl version installed by Linxubrew, do you happen to have any idea about that?
